test/src/lib/backoff.spec.ts

```diff
@@ -0,0 +1,136 @@
+import { expect } from 'chai';
+import { stub } from 'sinon';
+import rewire = require('rewire');
+
+import { StatusError } from '../../../src/lib/errors';
+
+const DEFAULT_OPTIONS = {
+	maxRetries: 5,
+	maxDelay: 900000, // 15 minutes
+	minDelay: 10000, // 10 seconds
+};
+
+const lib = rewire('../../../src/lib/backoff');
+
+describe('lib/backoff', async () => {
+	const withBackoff = lib.__get__('withBackoff');
+	const sleepStub = stub();
+
+	before(() => {
+		sleepStub.resolves();
+		lib.__set__('sleep', sleepStub);
+	});
+
+	beforeEach(() => {
+		sleepStub.reset();
+	});
+
+	it('resolves after 3 retries', async () => {
+		// Create a function that will fail 3 times so it succeeds on the 4th
+		const failer = new Failer(3);
```

Comment:
I see that the test failer module was deleted, are you waiting on discussions surrounding implementation before changing this code?